libraries/provider_component_runit_supervisor_upstart.rb

```diff
@@ -0,0 +1,71 @@
+require "chef/provider/lwrp_base"
+
+class Chef
+  class Provider
+    class ComponentRunitSupervisor
+      class Upstart < Chef::Provider::LWRPBase
+        provides :component_runit_supervisor, :platform_family => "debian"
+        provides :component_runit_supervisor, :platform_family => "rhel" do |node|
+          node['platform_version'] =~ /^6/
+        end
+        provides :component_runit_supervisor, :platform => "amazon"
+        provides :component_runit_supervisor, :platform => "fedora"
```

Comment:
Not sure what's correct, but what's implemented here is behaving the same as [what was there before](#diff-87632c3c644e61ff3846321eee72300bL27). This is a refactor so I'm preserving all previous behavior where not specified, even if it's wrong.